### PR TITLE
[fix] check if any data recalled by FilterModel

### PR DESCRIPTION
**Pull request by georgehristov**

### CODE REVIEW

File: src/Table/Column/FilterModel/TypeEnum.php [hunk 1]
```diff
@@ -30,15 +30,16 @@ public function init(): void
 
     public function setConditionForModel($model)
     {
-        $filter = $this->recallData();
-        $values = [];
-        foreach ($filter as $key => $isSet) {
-            if ($isSet === true) {
-                $values[] = $key;
+        if ($filter = $this->recallData()) {
```
Comment (mvorisek) [inline src/Table/Column/FilterModel/TypeEnum.php:33]: `[]` as default in FIlterMode::recallData?

Comment (georgehristov) [reply src/Table/Column/FilterModel/TypeEnum.php:33]: I was considering both options but this seems more robust I believe

Comment (mvorisek) [reply src/Table/Column/FilterModel/TypeEnum.php:33]: it seems fine, other checks looks like:
```

        $filter = $this->recallData();
        if (isset($filter['id'])) {
```

why can not we use the same?

Comment (mvorisek) [reply src/Table/Column/FilterModel/TypeEnum.php:33]: FIlterMode::recallData should definitely be declared with return type `?array`

**Review by mvorisek — APPROVED** [review]
see comments

File: src/Table/Column/FilterModel/TypeEnum.php (hunk 1 above)
Comment (georgehristov) [reply src/Table/Column/FilterModel/TypeEnum.php:33]: Maybe we do all 3, would not hurt.
And declare :array as return type.
Then it is robust :)

Comment (mvorisek) [reply src/Table/Column/FilterModel/TypeEnum.php:33]: if [] is good value for "not set" :) (not sure about this, probably not)

Comment (georgehristov) [reply src/Table/Column/FilterModel/TypeEnum.php:33]: `[]` is okay for this `isset($filter['id'])`

**Pull request merged**